src/ServiceInsight/Framework/Modules/CoreModule.cs

```diff
@@ -21,6 +21,7 @@ protected override void Load(ContainerBuilder builder)
             builder.RegisterType<ServiceControlConnectionProvider>().InstancePerLifetimeScope();
             builder.RegisterType<DefaultServiceControl>().As<IServiceControl>().InstancePerLifetimeScope();
             builder.RegisterType<CommandLineArgParser>().SingleInstance().OnActivating(e => e.Instance.Parse());
+            builder.RegisterType<WorkNotifier>().As<IWorkNotifier>().SingleInstance();
```

Comment:
Since we don't create child containers or lifetime scopes, wouldn't the effective lifetime scope of `InstancePerLifetimeScope` would be Singleton? Unless we use `BeginLifetimeScope` to resolve them, they won't get disposed, according to [this](http://docs.autofac.org/en/latest/lifetime/index.html).